src/corehaptics.cs

```diff
@@ -0,0 +1,398 @@
+		NSString TimeKey { get; }
+
+		[Field ("CHHapticPatternKeyEventDuration")]
+		NSString EventDurationKey { get; }
+
+		[Field ("CHHapticPatternKeyEventWaveformPath")]
+		NSString EventWaveformPathKey { get; }
+
+		[Field ("CHHapticPatternKeyEventParameters")]
+		NSString EventParametersKey { get; }
+
+		[Field ("CHHapticPatternKeyParameter")]
+		NSString ParameterKey { get; }
+
+		[Field ("CHHapticPatternKeyParameterID")]
+		NSString ParameterIdKey { get; }
+
+		[Field ("CHHapticPatternKeyParameterValue")]
+		NSString ParameterValueKey { get; }
+
+		[Field ("CHHapticPatternKeyParameterCurve")]
+		NSString WeakCHHapticParameterCurveKey { get; }
+
+		[Field ("CHHapticPatternKeyParameterCurveControlPoints")]
+		NSString WeakCHHapticParameterCurveControlPointsKey { get; }
+	}
+
+	[Mac (10,15), iOS (13,0)]
+	[StrongDictionary ("CHHapticPatternDefinitionKeys")]
+	partial interface CHHapticPatternDefinition {
+		double Version { get; set; }
+		NSArray Pattern { get; set; }
+		NSDictionary Event { get; set; }
+		CHHapticEventType EventType { get; set; }
+		double Time { get; set; }
+		double EventDuration { get; set; }
+		NSString EventWaveformPath { get; set; }
+		NSArray EventParameters { get; set; }
+		NSDictionary Parameter { get; set; }
+		// we need to do a NSString because it can be a CHHapticEventParameterID or a CHHapticDynamicParameterID 
+		// which are two different enums. User will have to do the conversion
+		[Advice ("Value can be either a CHHapticEventParameterId or a CHHapticDynamicParameterId. A check is needed against both enums.")]
+		NSString ParameterId { get; set; }
+		double ParameterValue { get; set; }
+		// we do not have docs or header information about the exact type
+		NSObject WeakCHHapticParameterCurve { get; set; }
+		NSObject WeakCHHapticParameterCurveControlPoints { get; set; }
+	}
+
+	[Mac (10,15), iOS (13,0)]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface CHHapticPattern {
+		[Export ("duration")]
+		double Duration { get; }
+
+		[Export ("initWithEvents:parameters:error:")]
+		IntPtr Constructor (CHHapticEvent[] events, CHHapticDynamicParameter[] parameters, [NullAllowed] out NSError outError);
+
+		[Export ("initWithEvents:parameterCurves:error:")]
+		IntPtr Constructor (CHHapticEvent[] events, CHHapticParameterCurve[] parameterCurves, [NullAllowed] out NSError outError);
+
+		[Export ("initWithDictionary:error:")]
+		IntPtr Constructor (NSDictionary patternDict, [NullAllowed] out NSError outError);
+
+		[Wrap ("this (patternDefinition?.Dictionary, out outError)")]
+		IntPtr Constructor (CHHapticPatternDefinition patternDefinition, [NullAllowed] out NSError outError);
+
+		[Export ("exportDictionaryAndReturnError:")]
+		[return: NullAllowed]
+		NSDictionary<NSString, NSObject> _ExportDictionary ([NullAllowed] out NSError outError);
+
+		[Wrap ("new CHHapticPatternDefinition (_ExportDictionary (out outError))")]
+		CHHapticPatternDefinition ExportDictionary ([NullAllowed] out NSError outError);
```

Comment:
but here you're not exporting a dictionary :) -> `Export`

Reply:
Fair enough, ExportAsPatternDefinition?

Reply:
`[Export]` is enough, the IDE gives you all other information